pkg/hintrunner/zero/zerohint_memcpy.go

```diff
@@ -28,3 +33,98 @@ func createVMExitScopeHinter(resolver hintReferenceResolver) (hinter.Hinter, err
 		},
 	}, nil
 }
+
+func createSearchSortedLowerHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	array_ptr, err := resolver.GetResOperander("array_ptr")
+	if err != nil {
+		return nil, err
+	}
+	elm_size, err := resolver.GetResOperander("elm_size")
+	if err != nil {
+		return nil, err
+	}
+	n_elms, err := resolver.GetResOperander("n_elms")
+	if err != nil {
+		return nil, err
+	}
+	key, err := resolver.GetResOperander("key")
+	if err != nil {
+		return nil, err
+	}
+	index, err := resolver.GetResOperander("index")
+	if err != nil {
+		return nil, err
+	}
+	return newSearchSortedLowerHint(array_ptr, elm_size, n_elms, key, index), nil
+}
+
+func newSearchSortedLowerHint(array_ptr, elm_size, n_elms, key, index hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SearchSortedLower",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+
+			//> array_ptr = ids.array_ptr
+			array_ptr, err := hinter.ResolveAsFelt(vm, array_ptr)
+			if err != nil {
+				return err
+			}
+
+			//> elm_size = ids.elm_size
+			elm_size, err := hinter.ResolveAsFelt(vm, elm_size)
+
+			//> assert isinstance(elm_size, int) and elm_size > 0, \
+			//> 	f'Invalid value for elm_size. Got: {elm_size}.'
+			if err != nil && utils.FeltLt(&utils.FeltZero, elm_size) {
+				return fmt.Errorf("Invalid value for elm_size. Got: %v.", elm_size)
+			}
+
+			//> n_elms = ids.n_elms
+			//> assert isinstance(n_elms, int) and n_elms >= 0, \
+			//> 	f'Invalid value for n_elms. Got: {n_elms}.'
+			n_elms, err := hinter.ResolveAsFelt(vm, n_elms)
+			if err != nil && utils.FeltLt(&utils.FeltZero, n_elms) {
+				return fmt.Errorf("Invalid value for n_elms. Got: %v.", n_elms)
+			}
+
+			var __find_element_max_size *fp.Element
+			// TODO: How to do this --> if __find_element_max_size is a variable?
+			//> if '__find_element_max_size' in globals():
+			//> 	assert n_elms <= __find_element_max_size, \
+			//> 		f'find_element() can only be used with n_elms<={__find_element_max_size}. ' \
+			//> 		f'Got: n_elms={n_elms}.'
+
+			if !utils.FeltLe(n_elms, __find_element_max_size) {
+				return fmt.Errorf("find_element() can only be used with n_elms<=%v. Got: n_elms=%v.", __find_element_max_size, n_elms)
+			}
+
+			key, err := hinter.ResolveAsFelt(vm, key)
+			if err != nil {
+				return err
+			}
+
+			indexAddr, err := index.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			//> for i in range(n_elms):
+			//> 	if memory[array_ptr + elm_size * i] >= ids.key:
+			//> 		ids.index = i
+			//> 		break
+			for i := uint64(0); i < n_elms.Uint64(); i++ {
+				v := memory.MemoryValueFromFieldElement(utils.Felt127.Add(array_ptr, elm_size))
```

Comment:
I belive, you have to resolve array_ptr as an address, add elem_size * i using ```array_ptr.AddOffset()```. Then resolve the read value as felt and compare it with key

Reply:
Thanks for the hand, makes lot of sense that method. Changed!